ddprof-lib/src/main/cpp/vmStructs.cpp

```diff
@@ -808,17 +808,22 @@ int NMethod::findScopeOffset(const void *pc) {
 }
 
 int ScopeDesc::readInt() {
-  unsigned char c = *_stream++;
-  unsigned int n = c - _unsigned5_base;
-  if (c >= 192) {
-    for (int shift = 6;; shift += 6) {
-      c = *_stream++;
-      n += (c - _unsigned5_base) << shift;
-      if (c < 192 || shift >= 24)
-        break;
+    unsigned char c = *_stream++;
+    if (c < _unsigned5_base) {
+        return 0;
     }
-  }
-  return n;
+    unsigned int n = c - _unsigned5_base;
+    if (c >= 192) {
+        for (int shift = 6; ; shift += 6) {
+            c = *_stream++;
+            if (c < _unsigned5_base) {
```

Comment:
is this not an indication of a race condition and we should retry the read ?
The previous byte indicated we were expecting something, so having 0 here is unexpected.

Reply:
reads of `NMethod`s are generally racy - we're reading them from a process-wide codecache which can change even when the sampled thread is halted in a signal handler.

Reply:
The more valid fix would then be to bail out.

Reply:
It seems that the problem is rather insidious - the `_unsigned5_base` is set up depending on the JVM version but that is subtly broken for Java 8 - see https://github.com/DataDog/java-profiler/issues/136

I am going to fix that problem first and if that does not stop the asan errors we can revisit this later.

Reply:
It seems that after #142 we are not getting to shifting a negative value.
I would keep this PR open for ~1 week to be sure the problem is fixed and then close it.